draft-ietf-webtrans-http2.md

```diff
@@ -186,31 +197,57 @@ the site that requested the creation of the session.
 established once it sends a 200 response. Both endpoints MUST NOT send any
-WebTransport frames on a given session before that session is established.
-
-## Limiting the Number of Simultaneous Sessions
-
-From a flow control perspective, WebTransport sessions count against HTTP/2
-session flow control limits just like regular HTTP requests, since they are
-established via an HTTP CONNECT request. This document does not make any effort
-to introduce a separate flow control mechanism for WebTransport sessions. If
-the server needs to limit the rate of incoming requests, it has alternative
-mechanisms at its disposal:
-
-* `HTTP_STREAM_REFUSED` error code defined in {{!RFC7540}} indicates to the
-  receiving HTTP/2 stack that the request was not processed in any way.
-* HTTP status code 429 indicates that the request was rejected due to rate
-  limiting {{!RFC6585}}. Unlike the previous method, this signal is directly
-  propagated to the application.
+WebTransport capsules on a given session before that session is established.
+
+## Flow Control
+
+### Limiting the Number of Simultaneous Sessions
+
+This document defines a SETTINGS_MAX_WEBTRANSPORT_SESSIONS parameter that allows
+the server to limit the maximum number of concurrent WebTransport sessions on a
+single HTTP/3 connection.  The client MUST NOT open more sessions than
+indicated in the server SETTINGS parameters.  The server MUST NOT close the
+connection if the client opens sessions exceeding this limit, as the client and
+the server do not have a consistent view of how many sessions are open due to
+the asynchronous nature of the protocol; instead, it MUST reset all of the
+CONNECT streams it is not willing to process with the `REFUSED_STREAM`
+error code defined in {{HTTP2}}.
+
+### Flow Control and Intermediaries {#flow-control-intermediaries}
+WebTransport over HTTP/2 uses several capsules for flow control, and all of
+these capsules define special intermediary handling as described in
+{{Section 3.2 of HTTP-DATAGRAM}}.  These capsules, referred to as the "flow
+control capsules" are WT_MAX_DATA, WT_MAX_STREAM_DATA, WT_MAX_STREAMS,
+WT_DATA_BLOCKED, WT_STREAM_DATA_BLOCKED, and WT_STREAMS_BLOCKED.
+
+Because flow control in WebTransport is hop-by-hop and does not provide an
+end-to-end signal, intermediaries MUST consume the flow control signals and
+express their own flow control limits to the next hop. The intermediary can
+send these signals via HTTP/3 flow control messages, HTTP/2 flow control
+messages, or as WebTransport flow control capsules, where appropriate.
+Intermediaries are responsible for storing any data for which they advertise
+flow control credit if that data cannot be immediately forwarded to the next
+hop.
+
+In practice, an intermediary that translates flow control signals between simlar
+WebTransport protocols, such as between two HTTP/2 connections, can often
+simply reexpress the same limits received on one connection directly on the
```

Comment:
Is reexpress a word?

Reply:
Heh, good question. Apparently, [yes](https://www.merriam-webster.com/dictionary/reexpress), although we can use the hyphenated form if preferable.